api/src/main/scala/com/lightbend/lagom/internal/api/ServiceReader.scala

```diff
@@ -142,6 +142,13 @@ object ServiceReader {
             }
           case _ => throw new IllegalStateException("ServiceCall does not have 3 type arguments?")
         }
+      } else if (ScalaSig.isScala(serviceInterface)) {
+        if (serviceInterface.isInterface()) {
+          val implClass = Class.forName(serviceInterface.getName + "$class", false, classLoader)
+          val method = implClass.getMethod(DescriptorMethodName, serviceInterface)
+          method.invoke(null, proxy.asInstanceOf[AnyRef])
+        } else
+          throw new IllegalArgumentException("Service.descriptor must be implemented in a trait")
```

Comment:
I don't understand how this worked - there is a branch in the if statement above that checks if the method being invoked is the `descriptor` method, and throws an exception:

```java
      } else if (method.equals(classOf[Service].getDeclaredMethod(DescriptorMethodName))) {
        throw new IllegalArgumentException("Service.descriptor must be implemented as a default method")
```

Since this check is done after that, it seems to me that the descriptor method on a Scala trait should never reach this point.